course_contents/2_intro_to_python/notes/02. Loops.md

```diff
@@ -0,0 +1,159 @@
+### Loops
+- loops are used to run blocks of code multiple times
+- there are two kinds of loops in Python:
+    1. while loops
+    2. for loops
+
+**Extra resources**
+[What is a loop?](https://python.tecladocode.com/3_continuous_countries/1_loops.html#what-is-a-loop)
+
+### The `while` loop
+- while loop runs an indefinite number of times i.e. while loop runs until the given conditional becomes False
+- syntax for writing the while loop:
+    1. use the `while` keyword at the beginning of the loop
+    2. the keyword should be followed with a correctly written conditional with a colon at the end of it
+    3. indented block of code that will run until the loop breaks
+
+An example of the properly written `while` loop:
+
+```py
+counter = 10
+while counter > 0:
+  print(f"{counter} seconds left until New Year!")
+  counter -= 1
+print("Happy New Year!")
+```
+
+An example of the `while` loop that will run until the program is manually stopped:
+
+```py
+counter = 10
+while counter > 0: # the counter never gets changed, causing the conditional to be always True
+  print(f"{counter} seconds left until New Year!")
+```
+
+### The `for` loop
+- for loop runs a given number of times
+- general syntax for `for` loop is:
+
+```py
+for variable in iterable:
```

Comment:
I would also mention the use of `_` as a variable name, and that this variable name is temporary to this for loop run time.

Reply:
I would say here that the `_` is a slightly more advanced edge-case, and we don't need to include it in this beginner-level cheatsheet.